Rebase for the latest version
Dongxiao Xu <dongxiao.xu@intel.com>

diff -ruN opkg-0.1.8-orig/libbb/unarchive.c opkg-0.1.8/libbb/unarchive.c
--- opkg-0.1.8-orig/libbb/unarchive.c	2010-07-20 09:39:02.266424893 +0800
+++ opkg-0.1.8/libbb/unarchive.c	2010-07-20 09:39:50.474435569 +0800
@@ -523,6 +523,10 @@
                 }
         }
 
+        if (strlen(tar_entry->name) > 100) {
+                tar_entry->name[100] = 0;
+        }
+
 	// tar_entry->name = xstrdup(tar.formated.name);
 
 /*
